drivers/platform/surface/surface_platform_profile.c

```diff
@@ -30,6 +39,7 @@ struct ssam_tmp_profile_info {
 struct ssam_tmp_profile_device {
 	struct ssam_device *sdev;
 	struct platform_profile_handler handler;
+	bool has_fan;
```

Comment:
Technically, we could call `device_property_read_bool` each time the profile is changed, but storing it as a local setting felt reasonable.

Reply:
Yeah, I think it's generally better to read properties once and store them.